examples/data_view/column_example.py

```diff
@@ -0,0 +1,188 @@
+
+from column_data_model import (
+    AbstractRowInfo, ColumnDataModel, HasTraitsRowInfo
+)
+
+
+class Address(HasStrictTraits):
+
+    street = Str
+
+    city = Str
+
+    country = Str
+
+
+class Person(HasStrictTraits):
+
+    name = Str
+
+    age = Int
+
+    address = Instance(Address)
+
+
+row_info = HasTraitsRowInfo(
+    title='People',
+    value='name',
+    value_type=TextValue(),
+    rows=[
+        HasTraitsRowInfo(
+            title="Age",
+            value="age",
+            value_type=IntValue(minimum=0),
+        ),
+        HasTraitsRowInfo(
+            title="Address",
+            value_type=no_value,
+            value='address',
+            rows=[
+                HasTraitsRowInfo(
+                    title="Street",
+                    value="address.street",
+                    value_type=TextValue(),
+                ),
+                HasTraitsRowInfo(
+                    title="City",
+                    value="address.city",
+                    value_type=TextValue(),
+                ),
+                HasTraitsRowInfo(
+                    title="Country",
+                    value="address.country",
+                    value_type=TextValue(),
+                ),
+            ],
+        ),
+    ],
+)
+
+
+class MainWindow(ApplicationWindow):
+    """ The main application window. """
+
+    data = List(Instance(Person))
+
+    row_info = Instance(AbstractRowInfo)
+
+    data_view = Instance(IDataViewWidget)
+
+    def _create_contents(self, parent):
+        """ Creates the left hand side or top depending on the style. """
+
+        self.data_view = DataViewWidget(
+            parent=parent,
+            data_model=ColumnDataModel(
+                data=self.data,
+                row_info=self.row_info,
+            ),
+        )
+        self.data_view._create()
+        return self.data_view.control
+
+    def _data_default(self):
+        import numpy
+        return numpy.random.uniform(size=(100000, 10))
+
+
+male_names = [
+    'Michael', 'Edward', 'Timothy', 'James', 'George', 'Ralph', 'David',
+    'Martin', 'Bryce', 'Richard', 'Eric', 'Travis', 'Robert', 'Bryan',
+    'Alan', 'Harold', 'John', 'Stephen', 'Gael', 'Frederic', 'Eli', 'Scott',
+    'Samuel', 'Alexander', 'Tobias', 'Sven', 'Peter', 'Albert', 'Thomas',
+    'Horatio', 'Julius', 'Henry', 'Walter', 'Woodrow', 'Dylan', 'Elmer',
+]
+
+female_names = [
+    'Leah', 'Jaya', 'Katrina', 'Vibha', 'Diane', 'Lisa', 'Jean', 'Alice',
+    'Rebecca', 'Delia', 'Christine', 'Marie', 'Dorothy', 'Ellen', 'Victoria',
+    'Elizabeth', 'Margaret', 'Joyce', 'Sally', 'Ethel', 'Esther', 'Suzanne',
+    'Monica', 'Hortense', 'Samantha', 'Tabitha', 'Judith', 'Ariel', 'Helen',
+    'Mary', 'Jane', 'Janet', 'Jennifer', 'Rita', 'Rena', 'Rianna',
+]
+
+all_names = male_names + female_names
+any_name = partial(choice, all_names)
+age = partial(randint, 15, 72)
+
+
+def family_name():
+    return choice([
```

Comment:
This stuff was copied form a TraitsUI table example (and cleaned up).  It would be useful to have code like this which generates random "people" in a common place for examples to use, but that is orthogonal to this PR.